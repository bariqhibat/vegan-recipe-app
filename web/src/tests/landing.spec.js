import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { render, waitFor } from "@testing-library/react";
import ApolloMockedProvider from "./utils/provider";

import Landing from "../routes/landing";

const jestMock = jest.fn();

describe("Landing", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  describe("Accessability", () => {
    test("Snapshot", async () => {
      const { asFragment } = render(
        <ApolloMockedProvider>
          <Router>
            <Landing jestMock={jestMock} />
          </Router>
        </ApolloMockedProvider>
      );

      await waitFor(() => expect(asFragment()).toMatchSnapshot());
    });
    test("Mock", async () => {
      render(
        <ApolloMockedProvider showErrors={false}>
          <Router>
            <Landing jestMock={jestMock} />
          </Router>
        </ApolloMockedProvider>
      );

      expect(jestMock).toHaveBeenCalled();
      await waitFor(() =>
        expect(jestMock).toHaveBeenLastCalledWith("landing rendered!")
      );
    });
  });
});
