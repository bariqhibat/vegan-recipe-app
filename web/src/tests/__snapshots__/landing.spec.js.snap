// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Accessability Snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      style="background-color: rgb(52, 73, 94); color: rgb(255, 255, 255); padding: 10rem;"
    >
      <div
        class="ant-row ant-row-space-between"
        style="margin-left: -4px; margin-right: -4px; height: 20rem;"
      >
        <div
          class="ant-col ant-col-xs-24 ant-col-md-24 ant-col-lg-12"
          style="padding-left: 4px; padding-right: 4px;"
        >
          <h1
            class="ant-typography"
            style="color: rgb(255, 255, 255);"
          >
            LAVA
          </h1>
          <h2
            class="ant-typography"
            style="color: rgb(255, 255, 255);"
          >
            Recipe for everyone
          </h2>
          <h1
            style="color: rgb(255, 255, 255);"
          >
            Be a pioneer to enlighten millions of people
          </h1>
          <h1
            style="color: rgb(255, 255, 255);"
          >
            Start on adding your recipes 
            <strong>
              now
            </strong>
            !
          </h1>
        </div>
        <div
          class="ant-col ant-col-xs-0 ant-col-md-0 ant-col-lg-12"
          style="padding-left: 4px; padding-right: 4px;"
        >
          <img
            alt="hero image"
            src="http://keyword-hero.com/wp-content/uploads/2017/04/Cart-Hero.png"
            style="height: 100%; position: absolute; right: 0px;"
          />
        </div>
      </div>
    </div>
    <div
      style="margin: auto; padding: 3rem 1rem 0px 1rem; text-align: center;"
    >
      <h2
        class="ant-typography"
      >
        Our current recipes
      </h2>
      <div
        style="margin: auto; padding: 1rem;"
      >
        <div
          class="ant-row"
        >
          <div
            class="ant-col ant-col-sm-24 ant-col-md-12 ant-col-lg-8"
          >
            <div
              style="text-align: center; min-width: 20rem; max-width: 20rem; margin: auto; padding: 1rem 1rem;"
            >
              <div
                class="ant-card ant-card-bordered ant-card-hoverable"
              >
                <div
                  class="ant-card-cover"
                >
                  <div
                    class="ant-skeleton ant-skeleton-element"
                  >
                    <div
                      class="ant-skeleton-image"
                      style="margin: 2rem 0px; width: 15rem; height: 15rem; min-height: 100%; min-width: 100%;"
                    >
                      <svg
                        class="ant-skeleton-image-svg"
                        viewBox="0 0 1098 1024"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          class="ant-skeleton-image-path"
                          d="M365.714286 329.142857q0 45.714286-32.036571 77.677714t-77.677714 32.036571-77.677714-32.036571-32.036571-77.677714 32.036571-77.677714 77.677714-32.036571 77.677714 32.036571 32.036571 77.677714zM950.857143 548.571429l0 256-804.571429 0 0-109.714286 182.857143-182.857143 91.428571 91.428571 292.571429-292.571429zM1005.714286 146.285714l-914.285714 0q-7.460571 0-12.873143 5.412571t-5.412571 12.873143l0 694.857143q0 7.460571 5.412571 12.873143t12.873143 5.412571l914.285714 0q7.460571 0 12.873143-5.412571t5.412571-12.873143l0-694.857143q0-7.460571-5.412571-12.873143t-12.873143-5.412571zM1097.142857 164.571429l0 694.857143q0 37.741714-26.843429 64.585143t-64.585143 26.843429l-914.285714 0q-37.741714 0-64.585143-26.843429t-26.843429-64.585143l0-694.857143q0-37.741714 26.843429-64.585143t64.585143-26.843429l914.285714 0q37.741714 0 64.585143 26.843429t26.843429 64.585143z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-card-body"
                >
                  <div
                    class="ant-card-meta"
                  >
                    <div
                      class="ant-card-meta-detail"
                    >
                      <div
                        class="ant-card-meta-description"
                      >
                        <div
                          class="ant-skeleton"
                        >
                          <div
                            class="ant-skeleton-content"
                          >
                            <h3
                              class="ant-skeleton-title"
                              style="width: 38%;"
                            />
                            <ul
                              class="ant-skeleton-paragraph"
                            >
                              <li />
                              <li />
                              <li
                                style="width: 61%;"
                              />
                            </ul>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-col ant-col-sm-24 ant-col-md-12 ant-col-lg-8"
          >
            <div
              style="text-align: center; min-width: 20rem; max-width: 20rem; margin: auto; padding: 1rem 1rem;"
            >
              <div
                class="ant-card ant-card-bordered ant-card-hoverable"
              >
                <div
                  class="ant-card-cover"
                >
                  <div
                    class="ant-skeleton ant-skeleton-element"
                  >
                    <div
                      class="ant-skeleton-image"
                      style="margin: 2rem 0px; width: 15rem; height: 15rem; min-height: 100%; min-width: 100%;"
                    >
                      <svg
                        class="ant-skeleton-image-svg"
                        viewBox="0 0 1098 1024"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          class="ant-skeleton-image-path"
                          d="M365.714286 329.142857q0 45.714286-32.036571 77.677714t-77.677714 32.036571-77.677714-32.036571-32.036571-77.677714 32.036571-77.677714 77.677714-32.036571 77.677714 32.036571 32.036571 77.677714zM950.857143 548.571429l0 256-804.571429 0 0-109.714286 182.857143-182.857143 91.428571 91.428571 292.571429-292.571429zM1005.714286 146.285714l-914.285714 0q-7.460571 0-12.873143 5.412571t-5.412571 12.873143l0 694.857143q0 7.460571 5.412571 12.873143t12.873143 5.412571l914.285714 0q7.460571 0 12.873143-5.412571t5.412571-12.873143l0-694.857143q0-7.460571-5.412571-12.873143t-12.873143-5.412571zM1097.142857 164.571429l0 694.857143q0 37.741714-26.843429 64.585143t-64.585143 26.843429l-914.285714 0q-37.741714 0-64.585143-26.843429t-26.843429-64.585143l0-694.857143q0-37.741714 26.843429-64.585143t64.585143-26.843429l914.285714 0q37.741714 0 64.585143 26.843429t26.843429 64.585143z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-card-body"
                >
                  <div
                    class="ant-card-meta"
                  >
                    <div
                      class="ant-card-meta-detail"
                    >
                      <div
                        class="ant-card-meta-description"
                      >
                        <div
                          class="ant-skeleton"
                        >
                          <div
                            class="ant-skeleton-content"
                          >
                            <h3
                              class="ant-skeleton-title"
                              style="width: 38%;"
                            />
                            <ul
                              class="ant-skeleton-paragraph"
                            >
                              <li />
                              <li />
                              <li
                                style="width: 61%;"
                              />
                            </ul>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-col ant-col-sm-24 ant-col-md-12 ant-col-lg-8"
          >
            <div
              style="text-align: center; min-width: 20rem; max-width: 20rem; margin: auto; padding: 1rem 1rem;"
            >
              <div
                class="ant-card ant-card-bordered ant-card-hoverable"
              >
                <div
                  class="ant-card-cover"
                >
                  <div
                    class="ant-skeleton ant-skeleton-element"
                  >
                    <div
                      class="ant-skeleton-image"
                      style="margin: 2rem 0px; width: 15rem; height: 15rem; min-height: 100%; min-width: 100%;"
                    >
                      <svg
                        class="ant-skeleton-image-svg"
                        viewBox="0 0 1098 1024"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          class="ant-skeleton-image-path"
                          d="M365.714286 329.142857q0 45.714286-32.036571 77.677714t-77.677714 32.036571-77.677714-32.036571-32.036571-77.677714 32.036571-77.677714 77.677714-32.036571 77.677714 32.036571 32.036571 77.677714zM950.857143 548.571429l0 256-804.571429 0 0-109.714286 182.857143-182.857143 91.428571 91.428571 292.571429-292.571429zM1005.714286 146.285714l-914.285714 0q-7.460571 0-12.873143 5.412571t-5.412571 12.873143l0 694.857143q0 7.460571 5.412571 12.873143t12.873143 5.412571l914.285714 0q7.460571 0 12.873143-5.412571t5.412571-12.873143l0-694.857143q0-7.460571-5.412571-12.873143t-12.873143-5.412571zM1097.142857 164.571429l0 694.857143q0 37.741714-26.843429 64.585143t-64.585143 26.843429l-914.285714 0q-37.741714 0-64.585143-26.843429t-26.843429-64.585143l0-694.857143q0-37.741714 26.843429-64.585143t64.585143-26.843429l914.285714 0q37.741714 0 64.585143 26.843429t26.843429 64.585143z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-card-body"
                >
                  <div
                    class="ant-card-meta"
                  >
                    <div
                      class="ant-card-meta-detail"
                    >
                      <div
                        class="ant-card-meta-description"
                      >
                        <div
                          class="ant-skeleton"
                        >
                          <div
                            class="ant-skeleton-content"
                          >
                            <h3
                              class="ant-skeleton-title"
                              style="width: 38%;"
                            />
                            <ul
                              class="ant-skeleton-paragraph"
                            >
                              <li />
                              <li />
                              <li
                                style="width: 61%;"
                              />
                            </ul>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-col ant-col-sm-24 ant-col-md-12 ant-col-lg-8"
          >
            <div
              style="text-align: center; min-width: 20rem; max-width: 20rem; margin: auto; padding: 1rem 1rem;"
            >
              <div
                class="ant-card ant-card-bordered ant-card-hoverable"
              >
                <div
                  class="ant-card-cover"
                >
                  <div
                    class="ant-skeleton ant-skeleton-element"
                  >
                    <div
                      class="ant-skeleton-image"
                      style="margin: 2rem 0px; width: 15rem; height: 15rem; min-height: 100%; min-width: 100%;"
                    >
                      <svg
                        class="ant-skeleton-image-svg"
                        viewBox="0 0 1098 1024"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          class="ant-skeleton-image-path"
                          d="M365.714286 329.142857q0 45.714286-32.036571 77.677714t-77.677714 32.036571-77.677714-32.036571-32.036571-77.677714 32.036571-77.677714 77.677714-32.036571 77.677714 32.036571 32.036571 77.677714zM950.857143 548.571429l0 256-804.571429 0 0-109.714286 182.857143-182.857143 91.428571 91.428571 292.571429-292.571429zM1005.714286 146.285714l-914.285714 0q-7.460571 0-12.873143 5.412571t-5.412571 12.873143l0 694.857143q0 7.460571 5.412571 12.873143t12.873143 5.412571l914.285714 0q7.460571 0 12.873143-5.412571t5.412571-12.873143l0-694.857143q0-7.460571-5.412571-12.873143t-12.873143-5.412571zM1097.142857 164.571429l0 694.857143q0 37.741714-26.843429 64.585143t-64.585143 26.843429l-914.285714 0q-37.741714 0-64.585143-26.843429t-26.843429-64.585143l0-694.857143q0-37.741714 26.843429-64.585143t64.585143-26.843429l914.285714 0q37.741714 0 64.585143 26.843429t26.843429 64.585143z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-card-body"
                >
                  <div
                    class="ant-card-meta"
                  >
                    <div
                      class="ant-card-meta-detail"
                    >
                      <div
                        class="ant-card-meta-description"
                      >
                        <div
                          class="ant-skeleton"
                        >
                          <div
                            class="ant-skeleton-content"
                          >
                            <h3
                              class="ant-skeleton-title"
                              style="width: 38%;"
                            />
                            <ul
                              class="ant-skeleton-paragraph"
                            >
                              <li />
                              <li />
                              <li
                                style="width: 61%;"
                              />
                            </ul>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-col ant-col-sm-24 ant-col-md-12 ant-col-lg-8"
          >
            <div
              style="text-align: center; min-width: 20rem; max-width: 20rem; margin: auto; padding: 1rem 1rem;"
            >
              <div
                class="ant-card ant-card-bordered ant-card-hoverable"
              >
                <div
                  class="ant-card-cover"
                >
                  <div
                    class="ant-skeleton ant-skeleton-element"
                  >
                    <div
                      class="ant-skeleton-image"
                      style="margin: 2rem 0px; width: 15rem; height: 15rem; min-height: 100%; min-width: 100%;"
                    >
                      <svg
                        class="ant-skeleton-image-svg"
                        viewBox="0 0 1098 1024"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          class="ant-skeleton-image-path"
                          d="M365.714286 329.142857q0 45.714286-32.036571 77.677714t-77.677714 32.036571-77.677714-32.036571-32.036571-77.677714 32.036571-77.677714 77.677714-32.036571 77.677714 32.036571 32.036571 77.677714zM950.857143 548.571429l0 256-804.571429 0 0-109.714286 182.857143-182.857143 91.428571 91.428571 292.571429-292.571429zM1005.714286 146.285714l-914.285714 0q-7.460571 0-12.873143 5.412571t-5.412571 12.873143l0 694.857143q0 7.460571 5.412571 12.873143t12.873143 5.412571l914.285714 0q7.460571 0 12.873143-5.412571t5.412571-12.873143l0-694.857143q0-7.460571-5.412571-12.873143t-12.873143-5.412571zM1097.142857 164.571429l0 694.857143q0 37.741714-26.843429 64.585143t-64.585143 26.843429l-914.285714 0q-37.741714 0-64.585143-26.843429t-26.843429-64.585143l0-694.857143q0-37.741714 26.843429-64.585143t64.585143-26.843429l914.285714 0q37.741714 0 64.585143 26.843429t26.843429 64.585143z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-card-body"
                >
                  <div
                    class="ant-card-meta"
                  >
                    <div
                      class="ant-card-meta-detail"
                    >
                      <div
                        class="ant-card-meta-description"
                      >
                        <div
                          class="ant-skeleton"
                        >
                          <div
                            class="ant-skeleton-content"
                          >
                            <h3
                              class="ant-skeleton-title"
                              style="width: 38%;"
                            />
                            <ul
                              class="ant-skeleton-paragraph"
                            >
                              <li />
                              <li />
                              <li
                                style="width: 61%;"
                              />
                            </ul>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-col ant-col-sm-24 ant-col-md-12 ant-col-lg-8"
          >
            <div
              style="text-align: center; min-width: 20rem; max-width: 20rem; margin: auto; padding: 1rem 1rem;"
            >
              <div
                class="ant-card ant-card-bordered ant-card-hoverable"
              >
                <div
                  class="ant-card-cover"
                >
                  <div
                    class="ant-skeleton ant-skeleton-element"
                  >
                    <div
                      class="ant-skeleton-image"
                      style="margin: 2rem 0px; width: 15rem; height: 15rem; min-height: 100%; min-width: 100%;"
                    >
                      <svg
                        class="ant-skeleton-image-svg"
                        viewBox="0 0 1098 1024"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          class="ant-skeleton-image-path"
                          d="M365.714286 329.142857q0 45.714286-32.036571 77.677714t-77.677714 32.036571-77.677714-32.036571-32.036571-77.677714 32.036571-77.677714 77.677714-32.036571 77.677714 32.036571 32.036571 77.677714zM950.857143 548.571429l0 256-804.571429 0 0-109.714286 182.857143-182.857143 91.428571 91.428571 292.571429-292.571429zM1005.714286 146.285714l-914.285714 0q-7.460571 0-12.873143 5.412571t-5.412571 12.873143l0 694.857143q0 7.460571 5.412571 12.873143t12.873143 5.412571l914.285714 0q7.460571 0 12.873143-5.412571t5.412571-12.873143l0-694.857143q0-7.460571-5.412571-12.873143t-12.873143-5.412571zM1097.142857 164.571429l0 694.857143q0 37.741714-26.843429 64.585143t-64.585143 26.843429l-914.285714 0q-37.741714 0-64.585143-26.843429t-26.843429-64.585143l0-694.857143q0-37.741714 26.843429-64.585143t64.585143-26.843429l914.285714 0q37.741714 0 64.585143 26.843429t26.843429 64.585143z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-card-body"
                >
                  <div
                    class="ant-card-meta"
                  >
                    <div
                      class="ant-card-meta-detail"
                    >
                      <div
                        class="ant-card-meta-description"
                      >
                        <div
                          class="ant-skeleton"
                        >
                          <div
                            class="ant-skeleton-content"
                          >
                            <h3
                              class="ant-skeleton-title"
                              style="width: 38%;"
                            />
                            <ul
                              class="ant-skeleton-paragraph"
                            >
                              <li />
                              <li />
                              <li
                                style="width: 61%;"
                              />
                            </ul>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
